### PR TITLE
DEP Bump minimum version of phpstan

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/.github/issues/349

Fixes https://github.com/silverstripe/silverstripe-standards/actions/runs/12342606211/job/34442740587

phpstan [1.12.6](https://github.com/phpstan/phpstan/releases/tag/1.12.6) seems to cause some things to be double counted, I'm not sure why

At a practical level, there's no real any harm in doing the double counting, and I have no idea how to stop it from happening since phpstan is what's doing the counting.

The other approach here is to simply lock the version to 1.12.5, though I'd prefer to just continue to use a caret.  We need to bump the minimum version so that the --prefer-lowest build continues to pass in CI

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,5 +28,11 @@
...
 	},
     "minimum-stability": "dev",
-    "prefer-stable": true
+    "prefer-stable": true,
+    "config": {
+        "allow-plugins": {
+            "composer/installers": true,
+            "silverstripe/vendor-plugin": true
+        }
+    }
```
Comment (ChloeSartorelli) [inline composer.json]: Why is this being added? Seems unrelated?

File: tests/PHPStan/KeywordSelfRuleTest.php [hunk 2]
```diff
@@ -24,17 +24,17 @@ public function provideRule()
             'interface' => [
                 'filePaths' => [__DIR__ . '/KeywordSelfRuleTest/TestInterface.php'],
                 'errorMessage' => "Can't use keyword 'self'. Use 'TestInterface' instead.",
-                'errorLines' => [13, 18, 18],
+                'errorLines' => [13, 13, 18, 18],
```
Comment (ChloeSartorelli) [inline tests/PHPStan/KeywordSelfRuleTest.php:28]: Add a comment here and for the others you've updated so if we look at this in the future and actually want to fix it, it's immediately clear what's incorrectly double and what's correct.
```suggestion
                // Note that 13 only actually has one violation, but does no harm to be counted twice.
                'errorLines' => [13, 13, 18, 18],
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
I suspect this is a result of running over both the `ClassConstFetch` (or other fetch/call) node _and_ the `Name` node. But probably no worth the time required to resolve it since like you say it's not actually doing any harm.

So long as we continue to catch errors and not have false negatives, I'm okay with it.

File: composer.json (hunk 1 above)
Comment (emteknetnz) [reply composer.json]: Removed

File: tests/PHPStan/KeywordSelfRuleTest.php (hunk 2 above)
Comment (emteknetnz) [reply tests/PHPStan/KeywordSelfRuleTest.php:28]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM
